### ISSUE TITLE
SourceTests.testBug206679_247198 fails

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse-platform/eclipse.platform.releng.aggregator#1419]
Side effect from https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/1384.
After merging of https://github.com/eclipse-equinox/equinox/pull/317 `SourceTests.testBug206679_247198`started to fail in I Builds.

I can see it locally too. This seem to be different from https://github.com/eclipse-pde/eclipse.pde/issues/779 because PDE complains about unsatisfied imports from `org.eclipse.jetty.*` packages in the `org.eclipse.help` feature, that should be there.

```
/data/git/eclipse.pde.ui/build/org.eclipse.pde.build/scripts/genericTargets.xml:112: Processing inclusion from feature org.eclipse.help: Bundle org.eclipse.equinox.http.jetty_3.9.0.v20231004-0455 failed to resolve.:
	Unsatisfied import package org.eclipse.jetty.http_[12.0.1,13.0.0).
	Unsatisfied import package org.eclipse.jetty.server_[12.0.1,13.0.0).
	Unsatisfied import package org.eclipse.jetty.server.handler_[12.0.1,13.0.0).
	Unsatisfied import package org.eclipse.jetty.session_[12.0.1,13.0.0).
	Unsatisfied import package org.eclipse.jetty.ee8.servlet_[12.0.1,13.0.0).
	Unsatisfied import package org.eclipse.jetty.ee8.nested_[12.0.1,13.0.0).
	Unsatisfied import package org.eclipse.jetty.util_[12.0.1,13.0.0).
	Unsatisfied import package org.eclipse.jetty.util.component_[12.0.1,13.0.0).
	Unsatisfied import package org.eclipse.jetty.util.ssl_[12.0.1,13.0.0).
	Unsatisfied import package org.eclipse.jetty.util.thread_[12.0.1,13.0.0).
```

Looking on the feature I see outdated bundles included:
![image](https://github.com/eclipse-pde/eclipse.pde/assets/964108/16abccdb-c1d5-4b3b-8ba2-293a14a98cb7)

I assume the only thing is missing is a "bump" in  `org.eclipse.help` feature so it "includes" new `org.eclipse.equinox.http.jetty` version.

I will push a PR.

**Comment by akurtakov** [comment]
IMHO this is https://github.com/eclipse-pde/eclipse.pde/issues/779 and not touching features.

**Comment by iloveeclipse** [comment]
Still the same problem, because the `org.eclipse.help` feature was **not** regenerated via https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1420.

@laeubi : is this some new tycho issue? How one can enforce feature to be regenerated (to pick **latest** versions of included plugins?

**Comment by iloveeclipse** [comment]
This is what is produced with latest I20231006-0120 build: 
![image](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/assets/964108/75d41df5-ecff-4b45-8651-bc4a151385c0)

**Comment by iloveeclipse** [comment]
It looks like tycho ignores changes on `forceQualifierUpdate.txt` now:

See [maven log](https://download.eclipse.org/eclipse/downloads/drops4/I20231006-0120/buildlogs/mb060_run-maven-build_output.txt) from https://download.eclipse.org/eclipse/downloads/drops4/I20231006-0120/buildlogs.php 

```
[INFO] -------------< org.eclipse.help.feature:org.eclipse.help >--------------
[INFO] Building org.eclipse.help.feature 2.3.1600-SNAPSHOT            [275/520]
[INFO]   from eclipse.platform.releng/features/org.eclipse.help-feature/pom.xml
[INFO] --------------------------[ eclipse-feature ]---------------------------
[INFO] Resolving dependencies of MavenProject: org.eclipse.help.feature:org.eclipse.help:2.3.1600-SNAPSHOT @ /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/pom.xml
[INFO] Maven Artifact org.eclipse.jdt.ui:org.eclipse.jdt.ui.junit.sampleproject:1.0.0-SNAPSHOT @ /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.jdt.ui/org.eclipse.jdt.ui.junit.sampleproject/target/org.eclipse.jdt.ui.junit.sampleproject-1.0.0-SNAPSHOT.jar is not a bundle and will be ignored, automatic wrapping of such artifacts can be enabled with <pomDependencies>wrapAsBundle</pomDependencies> in target platform configuration.
[WARNING] Problems resolving provisioning plan.: [Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet.http [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http.whiteboard 1.1.1.202109301733 to osgi.contract; (&(osgi.contract=JavaServlet)(version=3.1.0)).]
[WARNING] Problems resolving provisioning plan.: [Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet.http [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http.whiteboard 1.1.1.202109301733 to osgi.contract; (&(osgi.contract=JavaServlet)(version=3.1.0)).]
[WARNING] Problems resolving provisioning plan.: [Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet.http [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http.whiteboard 1.1.1.202109301733 to osgi.contract; (&(osgi.contract=JavaServlet)(version=3.1.0)).]
[WARNING] Problems resolving provisioning plan.: [Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet.http [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http.whiteboard 1.1.1.202109301733 to osgi.contract; (&(osgi.contract=JavaServlet)(version=3.1.0)).]
[WARNING] Problems resolving provisioning plan.: [Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet.http [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http.whiteboard 1.1.1.202109301733 to osgi.contract; (&(osgi.contract=JavaServlet)(version=3.1.0)).]
[WARNING] Problems resolving provisioning plan.: [Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http 1.2.2.202109301733 to java.package; javax.servlet.http [2.1.0,3.0.0).; Unable to satisfy dependency from org.osgi.service.http.whiteboard 1.1.1.202109301733 to osgi.contract; (&(osgi.contract=JavaServlet)(version=3.1.0)).]
[INFO] Downloading from eclipse.maven.central.mirror: https://repo.eclipse.org/content/repositories/maven_central/org/eclipse/jetty/jetty-util-ajax/12.0.1/jetty-util-ajax-12.0.1.pom
[INFO] Downloaded from eclipse.maven.central.mirror: https://repo.eclipse.org/content/repositories/maven_central/org/eclipse/jetty/jetty-util-ajax/12.0.1/jetty-util-ajax-12.0.1.pom (1.3 kB at 100 kB/s)
[INFO] 
[INFO] --- clean:3.3.1:clean (default-clean) @ org.eclipse.help ---
[INFO] Deleting /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/target
[INFO] 
[INFO] --- tycho-packaging:4.0.3:build-qualifier-aggregator (default-build-qualifier-aggregator) @ org.eclipse.help ---
[INFO] The project's OSGi version is 2.3.1600.v20231006-0120
[INFO] 
[INFO] --- tycho-packaging:4.0.3:validate-id (default-validate-id) @ org.eclipse.help ---
[INFO] 
[INFO] --- tycho-packaging:4.0.3:validate-version (default-validate-version) @ org.eclipse.help ---
[INFO] 
[INFO] --- resources:3.3.1:copy-resources (saveproperties) @ org.eclipse.help ---
[INFO] skip non existing resourceDirectory /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/saveproperties
[INFO] 
[INFO] --- enforcer:3.4.1:enforce (enforce-maven) @ org.eclipse.help ---
[INFO] 
[INFO] --- target-platform-configuration:4.0.3:target-platform (default-target-platform) @ org.eclipse.help ---
[INFO] 
[INFO] --- resources:3.3.1:resources (default-resources) @ org.eclipse.help ---
[INFO] skip non existing resourceDirectory /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/src/main/resources
[INFO] 
[INFO] --- resources:3.3.1:resources (filter-resources) @ org.eclipse.help ---
[INFO] skip non existing resourceDirectory /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/src/main/resources
[INFO] 
[INFO] --- tycho-apitools:4.0.3:generate (generate) @ org.eclipse.help ---
[INFO] 
[INFO] --- tycho-packaging:4.0.3:update-consumer-pom (default-update-consumer-pom) @ org.eclipse.help ---
[INFO] 
[INFO] --- tycho-source:4.0.3:plugin-source (plugin-source) @ org.eclipse.help ---
[INFO] 
[INFO] --- tycho-packaging:4.0.3:package-feature (default-package-feature) @ org.eclipse.help ---
[INFO] Building jar: /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/target/org.eclipse.help-2.3.1600-SNAPSHOT.jar
[INFO] 
[INFO] --- tycho-p2:4.0.3:p2-metadata-default (default-p2-metadata-default) @ org.eclipse.help ---
[INFO] 
[INFO] --- tycho-p2:4.0.3:feature-p2-metadata (default-feature-p2-metadata) @ org.eclipse.help ---
[INFO] 
[INFO] --- tycho-source:4.0.3:feature-source (feature-source) @ org.eclipse.help ---
[INFO] Building jar: /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/target/org.eclipse.help-2.3.1600-SNAPSHOT-sources-feature.jar
[INFO] 
[INFO] --- tycho-p2:4.0.3:p2-metadata (attach-p2-metadata) @ org.eclipse.help ---
[INFO] No baseline version MavenProject: org.eclipse.help.feature:org.eclipse.help:2.3.1600-SNAPSHOT @ /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/pom.xml
[INFO] 
[INFO] --- tycho-p2-extras:4.0.3:compare-version-with-baselines (compare-attached-artifacts-with-release) @ org.eclipse.help ---
[INFO] Execution was skipped
[INFO] 
[INFO] --- eclipse-jarsigner:1.4.2:sign (sign) @ org.eclipse.help ---
[INFO] Signing jar: /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/target/org.eclipse.help-2.3.1600-SNAPSHOT.jar
[INFO] Signing jar: /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/target/org.eclipse.help-2.3.1600-SNAPSHOT-sources-feature.jar
[INFO] 
[INFO] --- tycho-p2:4.0.3:p2-metadata (p2-metadata) @ org.eclipse.help ---
[INFO] No baseline version MavenProject: org.eclipse.help.feature:org.eclipse.help:2.3.1600-SNAPSHOT @ /home/jenkins/agent/workspace/Builds/I-build-4.30/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform.releng/features/org.eclipse.help-feature/pom.xml
[WARNING] artifact org.eclipse.help.feature:org.eclipse.help:xml:p2metadata:2.3.1600-SNAPSHOT already attached, replace previous instance
[WARNING] artifact org.eclipse.help.feature:org.eclipse.help:xml:p2artifacts:2.3.1600-SNAPSHOT already attached, replace previous instance
```

**Comment by laeubi** [comment]
> How one can enforce feature to be regenerated (to pick latest versions of included plugins?

A feature is always generated (except you do not a clean build), I also can't see that something is baseline replaced (the build even says it is not a baseline capable project). So either the build still sees the old versions or something else prevents the upload of the feature.

**Comment by laeubi** [comment]
As written above, nothing is replaced so you can touch this as much as you like without any effect:

> --- tycho-p2-extras:4.0.3:compare-version-with-baselines (compare-attached-artifacts-with-release) @ org.eclipse.help ---
> [INFO] Execution was skipped

So I can only assume that the outdated version are the ones that are actually replaced by an older version and probably require some kind of version bump but I have not et analyzed that in detail, at least you can see that for jetty 12 and http.servlet (I assume that something updated by @akurtakov recently) ar actually pulled in as "latest".

**Comment by iloveeclipse** [comment]
So it could be in fact PDE issue not properly resolving latest versions because of https://github.com/eclipse-pde/eclipse.pde/issues/779 . Let's wait for the fix then.

**Comment by HannesWell** [comment]
Maybe https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1430 already shows if this will be fixed with jetty 12.0.2.

**Comment by iloveeclipse** [comment]
https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1430 fixed the test. 

However, I still see "out of date" feature in I20231010-1800. I will create dedicated ticket for that.
